src/main/scala/firrtl/passes/CheckCombLoops.scala

```diff
@@ -169,8 +182,13 @@ object CheckCombLoops extends Pass {
       moduleGraphs(m.name) = DiGraph(internalDeps)
       simplifiedModuleGraphs(m.name) = moduleGraphs(m.name).simplify((m.ports map { p => LogicNode(p.name) }).toSet)
       for (scc <- moduleGraphs(m.name).findSCCs.filter(_.length > 1)) {
-        val cycle = recoverCycle(m.name,moduleGraphs,moduleDeps,Seq(m.name),scc :+ scc.head)
-        errors.append(new CombLoopException(m.info, m.name, cycle))
+        val sccSubgraph = moduleGraphs(m.name).subgraph(scc.toSet)
+        val cycle = findCycleInSCC(sccSubgraph)
+        (cycle zip cycle.tail) foreach {
```

Comment:
In Scala style, It used to be infix notation was reserved for purely functional things, foreach should always use a `.`. 

**But Scala changed it on me**. Now `.` notation is **always** recommended.